pkg/kubelet/kubelet.go

```diff
@@ -1853,6 +1862,7 @@ func (kl *Kubelet) HandlePodDeletions(pods []*api.Pod) {
 		if err := kl.deletePod(pod.UID); err != nil {
 			glog.V(2).Infof("Failed to delete pod %q, err: %v", kubeletUtil.FormatPodName(pod), err)
 		}
+		kl.probeManager.RemovePod(pod)
```

Comment:
You need to add this in HandlePodCleanup() too for the cases where pods were deleted during kubelet restarts. (kubelet doesn't have a checkpoint and it would not recognize that the pods were deleted at all).